zips/zip-0312.rst

```diff
@@ -126,60 +131,165 @@ Specification
+- The Spend authorizing key $\mathsf{ask}$ [#protocol-spendauthsig]_ is
+  replaced by the logical signing key that corresponds to the group public key
+  `PK`. By design, this key is never explicitly constructed (unless trusted
+  dealer key generation was used), and instead is represented by any sufficient
+  subset of the participant FROST signing key shares `sk_i`.
+
+The remaining parts of the Orchard and Sapling key trees SHOULD be generated
+from a common spending key $\mathsf{sk}$ as described in
+[#protocol-orchardkeycomponents]_ [#protocol-saplingkeycomponents]_, with the
+exception of $\mathsf{ask}$ which SHOULD NOT be derived from $\mathsf{sk}$ (and
+can't possibly be derived, if Distributed Key Generation was used).
+
+For both Orchard and Sapling, use of Distributed Key Generation precludes
+generating $\mathsf{ask}$ from a spending key $\mathsf{sk}. Instead the
+Distributed Key Generation protocol will output the FROST group public key `PK`.
+
+When generating FROST key components for the Orchard shielded protocol, the
+remaining parts of the Orchard key tree apart from $\mathsf{ask}$ and
+$\mathsf{ak}$ SHOULD be generated as described in
+[#draft-ecc-quantum-recoverability-usage-with-frost]_. The specification there
+is normative, and is only summarized here. It requires the participants to
+privately agree on a value $\mathsf{sk}$ (which is no longer sufficient by
+itself for spend authorization), and then use it to obtain the nullifier
+deriving key $\mathsf{nk}$, the quantum spending key $\mathsf{qsk}$, the quantum
+intermediate key $\mathsf{qk}$, and (using the FROST group public key in place
+of $\mathsf{ak}$) $\mathsf{rivk\_ext}$. This process is specified in § 4.2.3
+‘Orchard Key Components’ [#protocol-orchardkeycomponents]_ as modified by
+[#draft-ecc-quantum-recoverability-usage-with-frost]_.
+
+Similarly, when generating a FROST key for the Sapling shielded protocol, the
+remaining parts of the Sapling key tree apart from $\mathsf{ask}$ and
+$\mathsf{ak}$ (that is, $\mathsf{nk}$, $\mathsf{rivk\_ext}$, and further values
+derived from them and $\mathsf{ak}$) SHOULD be generated from a common value
+$\mathsf{sk}$ (again, not sufficient by itself for spend authorization) as
+described in [#protocol-saplingkeycomponents]_.
+
+If trusted dealer key generation is used contrary to recommendation, then
+$\mathsf{ask}$ MAY in that case be derived from an $\mathsf{sk}$ known only to
+the trusted dealer. This allows backing up the unsplit key, but any benefit of
+being able to do so does not outweigh the disadvantages of requiring a trusted
+dealer under normal circumstances.
+
+Regardless of how the Orchard or Sapling key components are generated, the
+participants MUST ensure that they have obtained the same values for
+$\mathsf{ak}$, $\mathsf{nk}$, and $\mathsf{rivk\_ext}$.
+
+Since Sapling will not be modified to support Quantum Recoverability
+[#draft-ecc-quantum-recoverability]_, Orchard is RECOMMENDED for new FROST keys.
```

Comment:
Note that `#draft-ecc-quantum-recoverability` provides the rationale for this choice.